src/ParkMate/ApplicationCore/ValueObjects/BookingPeriod.cs

```diff
@@ -11,28 +11,46 @@ public BookingPeriod(DateTime start, DateTime end)
         {
             Start = start;
             End = end;
+
+            if (start == end)
+                throw new ArgumentException("Start date and end date are equal");
+
+            if(end < start)
+                throw new ArgumentException("End date precedes start date");
         }
         public static BookingPeriod CreateOneHourPeriod(DateTime day)
         {
-            throw new NotImplementedException();
+            var start = day;
+            var end = day.AddHours(1);
+
+            return(new BookingPeriod(start, end));
```

Comment:
Creation of temporary start and end variables is redundant.
Return statements do not require braces.
This suggestion also applies to the other functions.
```suggestion
            return new BookingPeriod(day, day.AddHours(1));
```